velox/parse/Expressions.h

```diff
@@ -51,36 +52,26 @@ class Expressions {
     return resolverHook_;
   }
 
-  // Infers types for a single argument lambda. 'rowType' gives the
-  // names and types of columns available for capture. The lambda can
-  // be referenced with function(name) in a different expression.
-  static void registerLambda(
-      const std::string& name,
-      const std::shared_ptr<const RowType>& signature,
-      TypePtr rowType,
-      std::shared_ptr<const core::IExpr> body,
-      memory::MemoryPool* pool) {
-    auto types = rowType->as<TypeKind::ROW>().children();
-    types.insert(
-        types.end(),
-        signature->children().begin(),
-        signature->children().end());
-    auto names = rowType->as<TypeKind::ROW>().names();
-    names.insert(
-        names.end(), signature->names().begin(), signature->names().end());
-    auto lambdaRowType = ROW(std::move(names), std::move(types));
-    auto typedBody = inferTypes(body, lambdaRowType, pool);
-    lambdaRegistry()[name] =
-        std::make_shared<LambdaTypedExpr>(signature, typedBody);
-  }
-
  private:
-  static std::shared_ptr<const LambdaTypedExpr> lookupLambdaExpr(
-      std::shared_ptr<const IExpr> expr);
+  static TypedExprPtr inferTypes(
```

Comment:
Where is this function used? I'm wondering whether this PR can be split into two, one for parsing lambdas and the other for LambdaExpr::toSql()?

Reply:
@kagamiori This is a stack of 4 PRs / commits: #2652, #2655 and #2656

Reply:
Oh, gotcha. I was curious why there were several commits because we usually have one only.